go/client/cmd_account_recover_username.go

```diff
@@ -0,0 +1,74 @@
+package client
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+type CmdAccountRecoverUsername struct {
+	libkb.Contextified
+	email string
+	phone keybase1.PhoneNumber
+}
+
+func NewCmdAccountRecoverUsername(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:  "recover-username",
+		Usage: "Recover username via email or SMS.",
+		Action: func(c *cli.Context) {
+			cmd := NewCmdAccountRecoverUsernameRunner(g)
+			cl.ChooseCommand(cmd, "recover-username", c)
+		},
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "email",
+				Usage: "Email associated with the account.",
+			},
+			cli.StringFlag{
+				Name:  "phone",
+				Usage: "Phone number associated with the account including country code, e.g. +12344567890",
+			},
+		},
+	}
+}
+
+func NewCmdAccountRecoverUsernameRunner(g *libkb.GlobalContext) *CmdAccountRecoverUsername {
+	return &CmdAccountRecoverUsername{Contextified: libkb.NewContextified(g)}
+}
+
+func (c *CmdAccountRecoverUsername) ParseArgv(ctx *cli.Context) error {
+	c.email = ctx.String("email")
+	c.phone = keybase1.PhoneNumber(ctx.String("phone"))
+	if (len(c.email) > 0 && len(c.phone) > 0) || (len(c.email) == 0 && len(c.phone) == 0) {
```

Comment:
If you want to validate `c.phone`, there is a function: `libkb.IsPossiblePhoneNumber` that does a simple regexp to do entry-level validation on phone numbers, but it expects the format without `+` because it's used for SBS assertions.  It's fine to keep it though as is, but if it at least checked if it has a `+` and is numeric we could give a faster response to the user trying this command.

Reply:
that sounds good, none of the existing phone number handlers validated so I added checkers there as well.